core/src/main/java/nu/yona/server/device/entities/DeviceBase.java

```diff
@@ -60,12 +55,8 @@ public void setName(String name)
 
 	public boolean isVpnConnected()
 	{
-		return isVpnConnected;
-	}
-
-	public void setVpnConnected(boolean isVpnConnected)
-	{
-		this.isVpnConnected = isVpnConnected;
+		return getDeviceAnonymizedIfExisting().flatMap(DeviceAnonymized::getLastVpnStatusChangeEvent)
```

Comment:
I guess this is defensive programming that if the user is deleted, the buddy will not have load issues? Or is there a real situation in which the device anonymized can be `null`?

Reply:
Added this comment:

```
		// Call ..IfExisting, as we might be dealing with a buddy device here where the shared DeviceAnonymized has been deleted
		// by the buddy.
```